racket/collects/racket/contract/private/arr-i.rkt

```diff
@@ -583,18 +613,12 @@ evaluted left-to-right.)
                          [else
                           (values (cons (car kwds) kwds-rec)
                                   (cons (car kwd-args) args-rec))]))]))])
-    (cond
-      [(and res-checker (null? supplied-kwd-args))
-       (apply values res-checker
-              (append (filter (λ (x) (not (eq? x the-unsupplied-arg))) args) rest-args))]
-      [(null? supplied-kwd-args)
-       (apply values (append (filter (λ (x) (not (eq? x the-unsupplied-arg))) args) rest-args))]
-      [res-checker
-       (apply values res-checker supplied-kwd-args
-              (append (filter (λ (x) (not (eq? x the-unsupplied-arg))) args) rest-args))]
-      [else
-       (apply values supplied-kwd-args
-              (append (filter (λ (x) (not (eq? x the-unsupplied-arg))) args) rest-args))])))
```

Comment:
I'm not sure why I didn't write the simpler thing using the append here, but I'm skeptical of removing it.